modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java

```diff
@@ -55,113 +80,195 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
 
-        waitStrategy =
-            new WaitAllStrategy()
-                .withStrategy(Wait.forHttp("/ping").withBasicCredentials(username, password).forStatusCode(204))
-                .withStrategy(Wait.forListeningPort());
+        this.logger().info("Starting an InfluxDB container using [{}]", dockerImageName);
 
-        addExposedPort(INFLUXDB_PORT);
+        this.waitStrategy = new HttpWaitStrategy()
+            .forPath("/ping")
+            .withBasicCredentials(this.username, this.password)
+            .forStatusCode(NO_CONTENT_STATUS_CODE);
+
+        this.isAtLeastMajorVersion2 =
+            new ComparableVersion(dockerImageName.getVersionPart()).isGreaterThanOrEqualTo("2.0.0");
```

Comment:
@kiview I wonder if we can remove this check and use the version provided via ENV [here](https://github.com/influxdata/influxdata-docker/blob/8a3a2ae5bdae946cac8371a69cad45600830224f/influxdb/1.8/Dockerfile#L12). In order to do this we need to make a `dockerClient.inspectImageCmd(image)` at the configure method. WDYT?

Reply:
I will leave it as is for now. I think we need to think about how to address them for custom images.